magicctapipe/scripts/lst1_magic/lst1_magic_cloud_correction.py

```diff
@@ -0,0 +1,715 @@
+        clean_peak_time = peak_time[clean_mask]
+
+        # additional cleaning
+        if config["cloud_correction"]["use_additional_cleaning"]:
+            if tel_ids["LST-1"] == tel_id:
+
+                clean = tailcuts_clean(
+                    camgeom,
+                    image,
+                    boundary_thresh=config_clean_LST["boundary_thresh"],
+                    picture_thresh=config_clean_LST["picture_thresh"],
+                    min_number_picture_neighbors=config_clean_LST[
+                        "min_number_picture_neighbors"
+                    ],
+                )
+                # Create a mask indicating which pixels survived the cleaning
+                clean_mask = np.zeros_like(image, dtype=bool)
+                clean_mask[clean] = True
+
+                if np.sum(clean_mask) == 0:
+                    continue
+
+                # Apply the mask to relevant data arrays
+                clean_peak_time = peak_time[clean_mask]
+                clean_image = image[clean_mask]
+                clean_camgeom = camgeom[clean_mask]
+
+            else:
+
+                # Apply the image cleaning
+                clean_mask, image, peak_time = magic_clean.clean_image(
+                    event_image=image,
+                    event_pulse_time=peak_time,
+                    unsuitable_mask=unsuitable_mask,
+                )
+
+                # n_pixels = np.count_nonzero(clean_mask)
+                # n_islands, _ = number_of_islands(camgeom, clean_mask)
+
+                clean_camgeom = camgeom[clean_mask]  # camera_geom_masked
+                clean_image = image[clean_mask]  # image_masked
+                clean_peak_time = peak_time[clean_mask]  # signal_pixels
+
+                # Check if clean_image is empty or all zeros
+                if clean_image.size == 0 or not np.any(clean_image):
+                    continue  # Skip this iteration if clean_image is empty or all zeros
+
+        hillas_params = hillas_parameters(clean_camgeom, clean_image)
+        timing_params = timing_parameters(
+            clean_camgeom, clean_image, clean_peak_time, hillas_params
+        )
+        leakage_params = leakage_parameters(camgeom, image, clean_mask)
+        if tel_ids["LST-1"] == tel_id:
+            conc_params = concentration_parameters(
+                clean_camgeom, clean_image, hillas_params
+            )  # For LST-1 we compute concentration from the cleaned image and for MAGIC from the full image to reproduce the current behaviour in the standard code
+        else:
+            conc_params = concentration_parameters(camgeom, clean_image, hillas_params)
+
+        event_params = {
+            **hillas_params,
+            **timing_params,
+            **leakage_params,
+        }
+
+        prefixed_conc_params = {
+            f"concentration_{key}": value for key, value in conc_params.items()
+        }
+        event_params.update(prefixed_conc_params)
```

Comment:
prefixed_conc_params could not be included directly in event_params a few lines above?